Graphics/X11/Xft.hsc

```diff
@@ -190,14 +193,18 @@ foreign import ccall "XftFontClose"
 
 -- Support for multiple fonts --
 
-xftfont_ascent' :: [XftFont] -> IO Int
-xftfont_ascent' = (fmap maximum) . (mapM xftfont_ascent)
+foldIntNotEmpty :: (Int -> Int -> Int) -> Int -> [Int] -> Int
+foldIntNotEmpty _ def [] = def
+foldIntNotEmpty f _ l = foldl1' f l
 
-xftfont_descent' :: [XftFont] -> IO Int
-xftfont_descent' = (fmap maximum) . (mapM xftfont_descent)
+xftfont_max_ascent :: [XftFont] -> IO Int
+xftfont_max_ascent = (fmap $ foldIntNotEmpty max 0) . (mapM xftfont_ascent)
```

Comment:
Instead of `0` being a vague "something went wrong" value, I would push the responsibility to check this onto the caller, using `NonEmpty XftFont -> IO Int` here (alternatively we could return a `Maybe` or something)